tests/core/tools/promise.js

```diff
@@ -0,0 +1,60 @@
+/* bender-tags: editor */
+
+( function() {
+	'use strict';
+
+	bender.test( {
+
+		setUp: function() {
+			this.Promise = window.Promise;
+		},
+
+		tearDown: function() {
+			window.Promise = this.Promise;
+		},
+
+		// (#2962)
+		'test promise initialization': function() {
+			if ( window.Promise ) {
+				assert.areSame( window.Promise, CKEDITOR.tools.promise, 'Native Promise should be enabled' );
```

Comment:
Would it make sense to check

```js
assert.isUndefined( window.ES6Promise, 'Polyfill Promise should be loaded' );
```

here too?

Reply:
We can, to make sure polyfill is not loaded.